addons/binding/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/EEPHelper.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) 2014,2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.enocean.internal.eep;
+
+import javax.measure.quantity.Energy;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.enocean.internal.config.EnOceanChannelTotalusageConfig;
+
+/**
+ *
+ * @author Dominik Vorreiter - initial contribution
+ *
+ */
+public abstract class EEPHelper {
+    public static State validateTotalUsage(State value, State currentState, Configuration config) {
+        EnOceanChannelTotalusageConfig c = config.as(EnOceanChannelTotalusageConfig.class);
+
+        if (c.validateValue && (value instanceof QuantityType) && (currentState instanceof QuantityType)) {
+            @SuppressWarnings("unchecked")
+            QuantityType<Energy> newValue = value.as(QuantityType.class);
+
+            if (newValue != null) {
+                newValue = newValue.toUnit(SmartHomeUnits.KILOWATT_HOUR);
+            }
+
+            @SuppressWarnings("unchecked")
+            QuantityType<Energy> oldValue = currentState.as(QuantityType.class);
+
+            if (oldValue != null) {
+                oldValue = oldValue.toUnit(SmartHomeUnits.KILOWATT_HOUR);
+            }
+
+            if ((newValue != null) && (oldValue != null)) {
+                if (newValue.compareTo(oldValue) < 0) {
+                    if ((oldValue.subtract(newValue).doubleValue() < 1.0)) {
```

Comment:
Is this logic correct? It now evaluates to true if `newValue` is larger than `oldValue` (substract returns a negative value, but implies an increase in the data) and this increase is <1.0. I would assume `newValue` and `oldValue` need to be switched to make this work as I read from the conversation. Where it should filter if `newValue` is less than `oldValue`.

Reply:
Hi, thanks for your feedback! When you have a look at the line above you see that I first check if `newValue` is smaller than `oldValue`. If this is true than `oldValue - newValue` is always positive. So I check if the difference is smaller than 1.0 kWh and if so, the value is discarded. On the other hand, when `newValue` is greater or even `oldValue` it is checked if the difference is greater than 10.0 kWh to filter out wrong peaks.

Reply:
Ah yes your right. Don't know what I was thinking, but it wasn't very logical 😀